GitHubBranches.ps1

```diff
@@ -1072,128 +1074,1140 @@ filter Remove-GitHubRepositoryBranchProtectionRule
+        Allows deletion of the protected branch by anyone with write access to the repository.
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
 
     .INPUTS
-        [PSCustomObject]
+        GitHub.Repository
+        GitHub.Branch
 
     .OUTPUTS
-        GitHub.Branch
+        GitHub.BranchPatternProtectionRule
+
+    .NOTES
+        Protecting a branch requires admin or owner permissions to the repository.
+
+    .EXAMPLE
+        New-GitHubRepositoryBranchPatternProtectionRule -OwnerName microsoft -RepositoryName PowerShellForGitHub -BranchName release/**/* -EnforceAdmins
+
+        Creates a branch protection rule for the 'release/**/*' branch pattern of the PowerShellForGithub repository
+        enforcing all configuration restrictions for administrators.
+
+    .EXAMPLE
+        New-GitHubRepositoryBranchPatternProtectionRule -Uri 'https://github.com/microsoft/PowerShellForGitHub' -BranchName master -RequiredApprovingReviewCount 1
+
+        Creates a branch protection rule for the master branch of the PowerShellForGithub repository
+        requiring one approving review.
 #>
-    [CmdletBinding()]
-    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseSingularNouns", "", Justification="Internal helper that is definitely adding more than one property.")]
+    [CmdletBinding(
+        PositionalBinding = $false,
+        SupportsShouldProcess,
+        DefaultParameterSetName = 'Elements')]
+    [OutputType({$script:GitHubBranchPatternProtectionRuleTypeName })]
     param(
+        [Parameter(ParameterSetName = 'Elements')]
+        [string] $OwnerName,
+
+        [Parameter(ParameterSetName = 'Elements')]
+        [string] $RepositoryName,
+
         [Parameter(
             Mandatory,
-            ValueFromPipeline)]
-        [AllowNull()]
-        [AllowEmptyCollection()]
-        [PSCustomObject[]] $InputObject,
+            ValueFromPipelineByPropertyName,
+            Position = 1,
+            ParameterSetName = 'Uri')]
+        [Alias('RepositoryUrl')]
+        [string] $Uri,
 
-        [ValidateNotNullOrEmpty()]
-        [string] $TypeName = $script:GitHubBranchTypeName
-    )
+        [string] $OrganizationName,
 
-    foreach ($item in $InputObject)
-    {
-        $item.PSObject.TypeNames.Insert(0, $TypeName)
+        [Parameter(
+            Mandatory,
+            Position = 2)]
+        [string] $BranchPatternName,
 
-        if (-not (Get-GitHubConfiguration -Name DisablePipelineSupport))
-        {
-            if ($null -ne $item.url)
-            {
-                $elements = Split-GitHubUri -Uri $item.url
-            }
-            else
-            {
-                $elements = Split-GitHubUri -Uri $item.commit.url
-            }
-            $repositoryUrl = Join-GitHubUri @elements
+        [string[]] $StatusChecks,
```

Comment:
Good catch.  I'd say go ahead then and update the parameters there too.  `New-GitHubRepositoryBranchProtectionRule` is a low-usage method (no calls in the past 30 days), and given how PowerShell matches parameters, existing callers will continue to work correctly since removing the `s` would still allow for a partial, non-ambiguous match.

Reply:
That's fine.  Opened #320 to track the technical debt.
Your list of parameter name changes looks complete and accurate from what I had in mind.